### DISCUSSION TITLE
Select an image from the desktop by using codeception framework

### DISCUSSION THREAD

**Discussion by cuchakma** [Codeception/Codeception#6121] [General]
Greetings Everyone,

I am basically executing acceptance testing on a WordPress plugin, now it has come to my attention that I am trying to select an image from my local machine(laptop), unfortunately, the framework is not being able to do that. Can I kindly get a hint, on how to select an image from my local machine. I am trying to test whether my image uploader works or not. This is a very important feature for me.

Thank You

**Comment by Naktibalda** [comment]
I guess that [attachFile ](https://codeception.com/docs/modules/WebDriver#attachFile) doesn't do what you expect, am I right?

**Reply by cuchakma** [reply]
absolutely not, as I mentioned. I want the framework to be able to select files from the local machine after I create a script. I want the framework to click the 'Upload button', select 'The image'. unfortunately, it does not do that @Naktibalda

**Reply by Naktibalda** [reply]
I don't think that WebDriver protocol permits anything like that.
If you can find code doing that with Selenium for another framework or programming language, I will help to make it work with Codeception.
